dask/array/routines.py

```diff
@@ -1039,6 +1039,18 @@ def roll(array, shift, axis=None):
     return result
 
 
+@implements(np.shape)
```

Comment:
Since this has the same function name for NumPy and Dask, adding `shape` to the `routines` imports in `dask.array.__init__.py` here

https://github.com/dask/dask/blob/fbc373741264244102efc25d53893d0d83e669a6/dask/array/__init__.py#L27

should be sufficient for the `__array_function__` protocol to behave as expected. (Apologies I didn't notice this before)

```suggestion
```